test/src/notebook/notebook/widget.spec.ts

```diff
@@ -73,11 +133,248 @@ describe('notebook/notebook/widget', () => {
         expect(widget.hasClass('jp-Notebook')).to.be(true);
       });
 
-      it('should create widgets for existing cells', () => {
+    });
+
+    describe('#model', () => {
+
+      it('should get the model for the widget', () => {
         let model = new NotebookModel();
-        model.fromJSON(DEFAULT_CONTENT);
         let widget = new NotebookRenderer(model, defaultRenderMime());
-        expect(widget.childCount()).to.be(6);
+        expect(widget.model).to.be(model);
+      });
+
+      it('should be read-only', () => {
+        let widget = createWidget();
+        expect(() => { widget.model = null; }).to.throwError();
+      });
+
+    });
+
+    describe('#rendermime', () => {
+
+      it('should be the rendermime instance used by the widget', () => {
+        let rendermime = defaultRenderMime();
+        let widget = new NotebookRenderer(new NotebookModel(), rendermime);
+        expect(widget.rendermime).to.be(rendermime);
+      });
+
+      it('should be read-only', () => {
+        let widget = createWidget();
+        expect(() => { widget.rendermime = null; }).to.throwError();
+      });
+
+    });
+
+    describe('#childAt()', () => {
+
+      it('should get the child widget at a specified index', (done) => {
+        let widget = createWidget();
+        widget.attach(document.body);
```

Comment:
The requests are handled in order, and this is made after the request from update message.